### PR TITLE
LPS-102659 Experiment info in Experiences

### PR DESCRIPTION
**Pull request by andresfulla**

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	[STATUS_SCHEDULED]: 'warning',
+	[STATUS_TERMINATED]: 'danger'
+};
+
+/**
+ * This component is wrapper of ClayLabel
+ * It takes a value that gets translated to a label style and rendered
+ */
+class ExperimentsLabel extends Component {
+	prepareStateForRender(state) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js]: Add completely -un-necessary jsdocs

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+const STATUS_FINISHED_NO_WINNER = 4;
+const STATUS_FINISHED_WINNER = 3;
+const STATUS_PAUSED = 5;
+const STATUS_RUNNING = 1;
+const STATUS_SCHEDULED = 7;
+const STATUS_TERMINATED = 6;
+
+export const statusToLabelDisplayType = status => STATUS_TO_TYPE[status];
+
+const STATUS_TO_TYPE = {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js]: Add completely -un-necessary jsdocs

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	 * @default undefined
+	 * @review
+	 * @type {number}
+	 */
+	value: Config.number().required()
+};
+
+Soy.register(ExperimentsLabel, templates);
+
+export {ExperimentsLabel};
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js]: This is not needed

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	 */
+	size: Config.string(),
+
+	/**
+	 * This prop is generated in the `prepareStateForRender` from the `value` prop
+	 * @default undefined
+	 * @review
+	 * @type {string}
+	 */
+	style: Config.string().internal(),
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js]: Private attributes start with _

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+const STATUS_COMPLETED = 2;
+const STATUS_DRAFT = 0;
+const STATUS_FINISHED_NO_WINNER = 4;
+const STATUS_FINISHED_WINNER = 3;
+const STATUS_PAUSED = 5;
+const STATUS_RUNNING = 1;
+const STATUS_SCHEDULED = 7;
+const STATUS_TERMINATED = 6;
+
+export const statusToLabelDisplayType = status => STATUS_TO_TYPE[status];
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js]: I'm not sure if we can have exports through the code, let's see https://github.com/liferay/liferay-frontend-guidelines/issues/21#issuecomment-539378621

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import Component from 'metal-component';
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js]: Sort this imports (didn't npm-scripts sort this?)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+class ExperimentsLabel extends Component {
+	prepareStateForRender(state) {
+		const {value} = state;
+
+		const style = statusToLabelDisplayType(value);
+
+		const elementClasses =
+			state.size === 'lg' ? 'label-lg text-capitalize' : '';
+
+		return {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js]: this is a little opinionated, but I would write this:

```js
{
  ...state,
  elementClasses: state.size === 'lg' ? 'label-lg text-capitalize' : '',
  style: statusToLabelDisplayType(state.value)
};
```

I think it clarifies what the function does

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+			...state,
+			elementClasses,
+			style
+		};
+	}
+}
+
+ExperimentsLabel.STATE = {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js]: jsdocs

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.es.js [hunk 9]
```diff
@@ -26,6 +26,8 @@ import {
 import getConnectedComponent from '../../store/ConnectedComponent.es';
 import {setIn} from '../../utils/FragmentsEditorUpdateUtils.es';
 import templates from './SegmentsExperienceSelector.soy';
+import 'clay-icon';
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.es.js]: Sort imports? (sourceformat?)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.soy [hunk 10]
```diff
@@ -192,11 +196,35 @@
 
 												{$experience.segmentsEntryName}
 											</span>
+											{if $experience.segmentsExperimentStatus}
+												<div>
+													<span class="font-weight-normal text-secondary">
+														{msg desc=""}test{/msg}
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.soy]: This is perfectly well written, but I hate it

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.soy [hunk 11]
```diff
@@ -192,11 +196,35 @@
...
+
+													{call com.liferay.layout.content.page.editor.web.ExperimentsLabel.render}
+														{param label: $experience.segmentsExperimentStatus.label /}
+														{param value: $experience.segmentsExperimentStatus.value /}
+													{/call}
+												</div>
+											{/if}
 										</span>
+
+								{if $experience.hasLockedSegmentsExperiment}
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.soy]: Wrong indentation (yeiiii, God bless Soy)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/segmentsExperiences.es.js [hunk 12]
```diff
@@ -283,13 +284,11 @@ function createSegmentsExperienceReducer(state, action) {
...
 				_switchLayoutDataList(nextState, segmentsExperienceId)
-					.then(newState =>
-						setIn(
-							newState,
-							['segmentsExperienceId'],
-							segmentsExperienceId
-						)
-					)
+					.then(newState => ({
+						...newState,
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/segmentsExperiences.es.js]: I liked setIn, but this is more beautiful T.T

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js (hunk 6 above)
Comment (wincent) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperimentsLabel.es.js]: The new version of eslint-config-liferay and liferay-npm-scripts will sort imports, but we can't update yet (blocked on that build-output-directory change that is still in flight).

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.soy (hunk 10 above)
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.soy]: I'm getting rid of it

**Pull request closed without merging**